src/AutoRest.CSharp/MgmtTest/TestCommon/Extensions.cs

```diff
@@ -0,0 +1,176 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using AutoRest.CSharp.Input;
+using AutoRest.CSharp.Mgmt.Output;
+using AutoRest.CSharp.Mgmt.AutoRest;
+using AutoRest.CSharp.Mgmt.Models;
+using System.Reflection.Metadata.Ecma335;
+using AutoRest.CSharp.MgmtTest.Models;
+
+namespace AutoRest.CSharp.MgmtTest.TestCommon
+{
+    internal static class Extensions
+    {
+        public static string RefScenarioDefinedVariablesToString(this string src, IEnumerable<string> scenarioVariables)
+        {
+            StringBuilder stringBuilder = new StringBuilder(src);
+            stringBuilder = stringBuilder.Replace(@"{", @"{{").Replace(@"}", @"}}");
+            foreach (var scenarioVariable in scenarioVariables)
+            {
+                stringBuilder.Replace($"$({scenarioVariable})", $"{{{scenarioVariable}}}");
+            }
+            return stringBuilder.ToString();
+        }
+
+        public static FormattableString? RefScenarioDefinedVariables(this string src, IEnumerable<string>? scenarioVariables)
+        {
+            if (src is null)
+            {
+                return null;
+            }
+            if (scenarioVariables is null)
+            {
+                return $"{src:L}";
+            }
+            return $"{src.RefScenarioDefinedVariablesToString(scenarioVariables):L}";
+        }
+
+        public static string MultipleStartingSpace(this string src, int mul)
+        {
+            var lines = src.Split('\n');
+            for (int i = 0; i < lines.Length; i++)
+            {
+                var space = lines[i].TakeWhile(Char.IsWhiteSpace).Count();
+                lines[i] = $"{new string(' ', space * mul)}{lines[i].TrimStart()}";
+            }
+            return String.Join('\n', lines);
+        }
```

Comment:
This method serves formatting?

Reply:
yes, to increase the armtemplate multiline string starting space from 2 to 4 (the same with our generated code)

https://github.com/changlong-liu/autorest.csharp/blob/20220222-signalr/test/TestProjects/MgmtSignalR/tests/Generated/Scenario/SignalRTest.cs#L49